src/FontProvider.php

```diff
@@ -13,96 +18,118 @@
+	private $cache;
+
+	/** @var string[][] */
+	private $data;
+
+	public function __construct( CacheInterface $cache ) {
+		$this->cache = $cache;
+	}
 
 	/**
-	 * return data about a font
-	 * @param string $id Font id
-	 * @return array
+	 * Return data about a font.
+	 * @param string $id Font family name.
+	 * @return string[]|null Array of style names (e.g. Bold) to font filenames. Or null if no info found.
 	 */
-	public static function getData( string $id ) {
-		if ( isset( self::$data[$id] ) ) {
-			return self::$data[$id];
-		} else {
+	public function getData( string $id ) {
+		if ( $id === '' ) {
 			return null;
 		}
+
+		if ( isset( $this->data[ $id ] ) ) {
+			return $this->data[ $id ];
+		}
+
+		$this->data = $this->cache->get( 'fonts', function ( ItemInterface $item ) {
+			$item->expiresAfter( new DateInterval( 'P7D' ) );
+			return $this->getFontData( array_keys( $this->getList() ) );
+		} );
+
+		return $this->data[ $id ] ?? null;
 	}
 
 	/**
-	 * return list of fonts
-	 * @return array
+	 * Get a list of locally installed fonts.
+	 * @return string[][]
 	 */
-	public static function getList() {
-		$list = [];
-		foreach ( self::$data as $key => $font ) {
-			$list[$key] = $font['label'];
+	private function getFontData( $fonts ): array {
+		$fontData = [];
+		foreach ( $fonts as $font ) {
+			if ( !trim( $font ) ) {
+				// Skip empty font name.
+				continue;
+			}
+			// Get semicolon-separated parts of font information.
+			$cmd = 'fc-list :family="' . $font . '" --format "%{file};%{family};%{style}\n"';
+			$process = Process::fromShellCommandline( $cmd );
+			$process->mustRun();
+			$output = $process->getOutput();
+
+			// Format the output into a single array.
+			$lines = array_filter( explode( "\n", $output ) );
+			foreach ( $lines as $line ) {
+				$parts = str_getcsv( $line, ';' );
+				if ( count( $parts ) !== 3 ) {
+					// Guard against malformed results.
+					continue;
+				}
+				$file = $parts[0];
+				$family = $parts[1];
+				$style = $parts[2];
+				if ( !isset( $fontData[$family] ) ) {
+					$fontData[$family] = [];
+				}
+				$fontData[$family][$style] = $file;
+			}
 		}
 
-		return $list;
+		return $fontData;
+	}
+
+	/**
+	 * Get the full list of available fonts.
+	 * @return string[]
+	 */
+	public function getList() {
+		return [
+			// Font family name => English label
+			'FreeSerif' => 'FreeSerif', // Hard-coded default for non-latin scripts.
+			'Linux Libertine' => 'Linux Libertine',
+			'Libertinus' => 'Libertinus',
+			'Mukta' => 'Mukta (Devanagari)',
+			'Mukta Mahee' => 'Mukta Mahee (Gurmukhi)',
+			'Mukta Malar' => 'Mukta Malar (Tamil)',
+			'Lohit Kannada' => 'Lohit Kannada (Kannada)',
+		];
 	}
 
 	/**
-	 * return CSS
+	 * Get CSS for the given font.
+	 * @param string $id The font ID as defined in this class.
 	 * @return string
 	 */
-	public static function getCss( $id, $basePath ) {
-		if ( !isset( self::$data[$id] ) ) {
+	public function getCss( string $id ): string {
```

Comment:
**Question**: This method used to assign all font families even if not defined/included. Now it seems to only do that for the one that is selected by the user. I'm guessing this new way is intentional and a bit of a cleanup?
I also see body defined twice in the final css. Not sure if that's a problem
```
body { font-family: "FreeSerif" }
body {
	line-height: 1.2;
}
```

Reply:
I think it doesn't really matter. Mentioned it in case it was an easy fix